crates/typed-store/src/rocks/mod.rs

```diff
@@ -1285,6 +1285,43 @@ impl<K, V> DBMap<K, V> {
         readopts
     }
 
+    /// Creates a safe reversed iterator with optional bounds.
+    /// Upper bound is included.
+    pub fn reversed_safe_iter_with_bounds(
+        &self,
+        lower_bound: Option<K>,
```

Comment:
(1) I see there is two approaches in DBMap for bounds - there is either `Option<>` where Some(...) means included bound, and there is approach in `range_iter` and `create_read_options_with_range` method where user can select type of bounds to be either included or excluded. 
Should we try to just adopt same style API for reverse iterator and name method `reverse_range_iter`? (e.g. allow user to select type of bounds). Especially since it looks like we are calling `create_read_options_with_range` internally anyway.

Reply:
maybe, we can further simplify and merge methods. We can do it as a followup though. This PR is already large/risky enough